### PR TITLE
Remove support for Python 2

### PR DESCRIPTION
**Pull request by ksindi**
- Remove support Python 2 which is EOL
- Add .travis.yml for CI

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	python setup.py install
+
+install-all:
+	pip install -e .[all]
+
+lint:
+	pytest --flake8 buildpipe tests
+
+test:
+	python setup.py test
```
Comment (pdumoulin) [inline Makefile]: For awhile there are going to be systems with python2 installed where that is the default python version. We may want to change these commands to explicitly use python3 or fail commands if python2 is default.

File: setup.py [hunk 2]
```diff
@@ -29,58 +29,39 @@ def get_version():
     raise RuntimeError('Unable to find version string.')
 
 
-class Tox(TestCommand):
```
Comment (pdumoulin) [inline setup.py:32]: After these changes, the import for "sys" and "TestCommand" are no longer necessary.

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+test-all:
```
Comment (pdumoulin) [inline Makefile]: I'm not sure if it is my setup or an actual issue, but I am unable to get all tests to run due to an error that appears to be related to the flags for test coverage.

```
py37 runtests: commands[0] | py.test tests
ERROR: usage: py.test [options] [file_or_dir] [file_or_dir] [...]
py.test: error: unrecognized arguments: --cov=jwplatform --cov-report
  inifile: /home/paul/projects/jwplatform-py/setup.cfg
  rootdir: /home/paul/projects/jwplatform-py

ERROR: InvocationError for command '/home/paul/projects/jwplatform-py/.tox/py37/bin/py.test tests' (exited with code 4)
```

File: .gitignore [hunk 4]
```diff
@@ -1,10 +1,104 @@
...
+nosetests.xml
+coverage.xml
+*.cover
+.hypothesis/
+
+# Translations
+*.mo
+*.pot
+
+# Django stuff:
```
Comment (pdumoulin) [inline .gitignore:58]: Many of the entries below this line don't seem relevant to this project. (Flask, Django and Jupyter)

File: Makefile (hunk 3 above)
Comment (pdumoulin) [reply Makefile]: I am able to run `test` but not `test-all`

Comment (pdumoulin) [reply Makefile]: Confirmed the CLI options work when running pytest directly, not sure why using tox is not working.

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+	tox
+
+version:
+	python setup.py --version
+
+docs:
```
Comment (pdumoulin) [inline Makefile]: `sphinx` is not listed in packages in `setup.py` - not sure if it needs to be there, but it doesn't get installed using `install-all`

File: Makefile [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+	rm -f docs/jwplatform.rst
+	rm -f docs/modules.rst
+	sphinx-apidoc -o docs/ buildpipe
```
Comment (pdumoulin) [inline Makefile]: Is there an installation/directory creation step for buildpipe that is needed here?

**Review by pdumoulin — COMMENTED** [review]
Might be good to have @rmnl who has more experience with this project and @TBoshoven who has done lots of python2 -> 3 work to review.

File: Makefile (hunk 6 above)
Comment (ksindi) [reply Makefile]: Will remove

File: Makefile (hunk 5 above)
Comment (ksindi) [reply Makefile]: I'll remove docs for now. Copy :spaghetti:.

File: jwplatform/client.py [hunk 7]
```diff
@@ -57,13 +47,13 @@ class Client(object):
         >>> jwplatform_client = jwplatform.Client('API_KEY', 'API_SECRET')
     """
 
-    def __init__(self, key, secret, *args, **kwargs):
+    def __init__(self, key: str, secret: str, *args, **kwargs):
         self.__key = key
         self.__secret = secret
 
         self._scheme = kwargs.get('scheme') or 'https'
         self._host = kwargs.get('host') or 'api.jwplatform.com'
```
Comment (rmnl) [inline jwplatform/client.py:55]: Unrelated to this update, but why are we not adding more arguments instead of trying to get them from the `kwargs` argument. It would give an IDE better hints if we just name the arguments.

File: setup.py [hunk 8]
```diff
@@ -29,58 +29,39 @@ def get_version():
...
         'Intended Audience :: Developers',
         'Topic :: Software Development :: Libraries :: Python Modules',
         'License :: OSI Approved :: MIT License',
-        'Programming Language :: Python :: 2',
-        'Programming Language :: Python :: 2.7',
         'Programming Language :: Python :: 3',
         'Programming Language :: Python :: 3.5',
         'Programming Language :: Python :: 3.6',
+        'Programming Language :: Python :: 3.7',
     ],
```
Comment (rmnl) [inline setup.py:51]: Should we add support for 3.8 too?

Comment (ksindi) [reply setup.py:51]: Makes sense. I'll also note it as Python 3 only.

File: jwplatform/client.py (hunk 7 above)
Comment (ksindi) [reply jwplatform/client.py:55]: 100%. I was thinking doing that in a follow up PR.

File: .gitignore (hunk 4 above)
Comment (ksindi) [reply .gitignore:58]: This is the standard gitignore from GitHub. https://github.com/github/gitignore/blob/master/Python.gitignore. Happy to slim down if you want.

Comment (pdumoulin) [reply .gitignore:58]: I don't have a strong feeling either way, as long as this was intentional.

**Review by polishmatt — APPROVED** [review]

File: Makefile [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+	pip install -e .[all]
+
+test:
+	python3 setup.py test
+
+version:
+	python3 setup.py --version
+
+release: clean build
+	python setup.py sdist upload
```
Comment (pdumoulin) [inline Makefile]: Should these commands also be changed to use python3 explicitly?

File: Makefile [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+	rm -rf build/
+	rm -rf dist/
+	rm -rf .eggs
+	rm -rf *.egg-info
+
+install: clean-build
+	python3 setup.py install
+
+install-all:
+	pip install -e .[all]
```
Comment (pdumoulin) [inline Makefile]: Same as changing python to python3 - should this be pip3?

File: setup.py [hunk 11]
```diff
@@ -29,58 +27,41 @@ def get_version():
...
         'Topic :: Software Development :: Libraries :: Python Modules',
         'License :: OSI Approved :: MIT License',
-        'Programming Language :: Python :: 2',
-        'Programming Language :: Python :: 2.7',
+        'Programming Language :: Python',
         'Programming Language :: Python :: 3',
         'Programming Language :: Python :: 3.5',
         'Programming Language :: Python :: 3.6',
+        'Programming Language :: Python :: 3.7',
+        'Programming Language :: Python :: 3.8',
```
Comment (pdumoulin) [inline setup.py:49]: 3.8 isn't listed in tox.ini or travis.yml - is it officially supported?
